Text/HealthcareTA/HealthcareTA.cs

```diff
@@ -0,0 +1,136 @@
+
+        [FunctionName("HealthcareTA")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log,
+            ExecutionContext executionContext)
+        {
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            // Get Endpoint and access key from App Settings
+            string apiKey = Environment.GetEnvironmentVariable(healthcareApiEnvKey, EnvironmentVariableTarget.Process);
+            string apiEndpoint = Environment.GetEnvironmentVariable(healthcareApiEnvEndpoint, EnvironmentVariableTarget.Process);
+            if (apiKey == null || apiEndpoint == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Healthcare Text Analytics API key or endpoint is missing. Make sure to set them in the Environment Variables.");
+            }
+
+            var client = new TextAnalyticsClient(new Uri(apiEndpoint), new AzureKeyCredential(apiKey));
+
+            // Get a custom timeout from the header, if it exists. If not use the default timeout.
+            int timeout;
+            if (!int.TryParse(req.Headers["SkillTimeout"].ToString(), out timeout))
+            {
+                timeout = defaultTimeout;
+            }
+            var timeoutMiliseconds = timeout * 1000;
+
+            WebApiSkillResponse response = await WebApiSkillHelpers.ProcessRequestRecordsAsync(skillName, requestRecords,
+                async (inRecord, outRecord) => {
+                    // Prepare analysis operation input
+                    var document = inRecord.Data["document"] as string;
+                    var options = new AnalyzeHealthcareEntitiesOptions { };
+                    if (document.Length >= maxCharLength)
+                    {
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract
+                        {
+                            Message = $"Healthcare Text Analytics Error: The submitted document was over {maxCharLength} characters. It has been truncated to fit this requirement."
+                        });
+                        document = document.Substring(0, maxCharLength);
+                    }
+                    List<string> batchInput = new List<string>()
+                    {
+                        document
+                    };
+
+                    // start analysis process
+                    var timer = System.Diagnostics.Stopwatch.StartNew();
+                    AnalyzeHealthcareEntitiesOperation healthOperation = await client.StartAnalyzeHealthcareEntitiesAsync(batchInput, "en", options);
```

Comment:
>await client.StartAnalyzeHealthcareEntitiesAsync(batchInput, "en", options); [](http://example.com/codeflow?start=73&length=76)

Oh something I just realized, this API supports passing a cancellationToken, so we might be able to just use that to control the timeout. Let's chat about what that means later today.

[https://docs.microsoft.com/en-us/dotnet/api/azure.ai.textanalytics.textanalyticsclient.startanalyzehealthcareentitiesasync?view=azure-dotnet-preview#Azure_AI_TextAnalytics_TextAnalyticsClient_StartAnalyzeHealthcareEntitiesAsync_System_Collections_Generic_IEnumerable_System_String__System_String_Azure_AI_TextAnalytics_AnalyzeHealthcareEntitiesOptions_System_Threading_CancellationToken](https://docs.microsoft.com/en-us/dotnet/api/azure.ai.textanalytics.textanalyticsclient.startanalyzehealthcareentitiesasync?view=azure-dotnet-preview#Azure_AI_TextAnalytics_TextAnalyticsClient_StartAnalyzeHealthcareEntitiesAsync_System_Collections_Generic_IEnumerable_System_String__System_String_Azure_AI_TextAnalytics_AnalyzeHealthcareEntitiesOptions_System_Threading_CancellationToken)_

Reply:
Created a work item for this: https://msdata.visualstudio.com/Azure%20Search/_workitems/edit/1251209